plugins/transport-nio/src/main/java/org/elasticsearch/transport/nio/NioNotEnabledBootstrapCheck.java

```diff
@@ -21,12 +21,23 @@
 
 import org.elasticsearch.bootstrap.BootstrapCheck;
 import org.elasticsearch.bootstrap.BootstrapContext;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+
+import static org.elasticsearch.common.settings.Setting.boolSetting;
 
 public class NioNotEnabledBootstrapCheck implements BootstrapCheck {
 
+    public static final Setting<Boolean> OVERRIDE_BOOTSTRAP =
```

Comment:
Tbh, I was not sure whether a system property or a setting would be a better choice but using a setting for this is consistent with what we did previously (see e.g. #20511).